widget/embedded/src/utils/quote.ts

```diff
@@ -326,3 +293,60 @@ export const getUniqueBlockchains = (steps: Step[]) => {
 
   return result;
 };
+
+export const sortRoutesBy = (
+  strategy: SortStrategy,
+  routes: MultiRouteResponse['results']
+): MultiRouteResponse['results'] => {
+  return routes.sort((route1, route2) => {
+    const sortPreferenceType = ROUTE_SORTING_STRATEGY[strategy];
+
+    const route1Score =
+      route1.scores?.find(
+        (score) => score.preferenceType === sortPreferenceType
+      )?.score ?? 0;
+    const route2Score =
+      route2.scores?.find(
+        (score) => score.preferenceType === sortPreferenceType
+      )?.score ?? 0;
+
+    if (sortPreferenceType === 'NET_OUTPUT' || sortPreferenceType === 'PRICE') {
+      return route2Score - route1Score;
+    }
+    return route1Score - route2Score;
+  });
+};
+// Function to get a unique key for each object
+const getKey = (obj: RouteTag) => obj.value;
+export function getUniqueTags(tags: RouteTag[]) {
+  // Create a Map to store unique objects based on their key
+  const uniqueObjectsMap = new Map();
+
+  tags.forEach((obj) => {
+    const key = getKey(obj);
+    uniqueObjectsMap.set(key, obj);
+  });
+  // Convert the Map values back to an array
+  return Array.from(uniqueObjectsMap.values());
+}
```

Comment:
Kindly include return types for the functions.

Reply:
fixed